modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.client/src/main/scala/edu/gemini/seqexec/web/client/components/sequence/steps/StepConfigTable.scala

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) 2016-2017 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package edu.gemini.seqexec.web.client.components.sequence.steps
+
+import scala.scalajs.js
+import edu.gemini.seqexec.model.Model.{Step, SystemName}
+import edu.gemini.seqexec.web.client.components.SeqexecStyles
+import japgolly.scalajs.react._
+import japgolly.scalajs.react.component.Scala.Unmounted
+import japgolly.scalajs.react.vdom.html_<^._
+
+import scalacss._
+import scalaz.std.AllInstances._
+import scalaz.syntax.traverse._
+import scalaz.syntax.equal._
+import react.virtualized._
+
+object StepConfigTable {
+  private val CssSettings = scalacss.devOrProdDefaults
+  import CssSettings._
+
+  final case class Props(step: Step, size: Size) {
+    val settingsList: List[(SystemName, String, String)] =
+      step.config.toList.flatMap {
+        case (s, c) =>
+          c.map {
+            case (k, v) => (s, k, v)
+          }
+      }
+
+    val rowCount: Int = settingsList.size
+
+    def rowGetter(idx: Int): SettingsRow =
+      settingsList.index(idx).fold(SettingsRow.Zero)(Function.tupled(SettingsRow.apply))
+  }
+
+  // ScalaJS defined trait
+  // scalastyle:off
+  trait SettingsRow extends js.Object {
+    var sub: SystemName
+    var name: String
+    var value: String
+  }
+  // scalastyle:on
+  object SettingsRow {
+    @SuppressWarnings(Array("org.wartremover.warts.AsInstanceOf"))
+    def apply(sub: SystemName, name: String, value: String): SettingsRow = {
+      val p = (new js.Object).asInstanceOf[SettingsRow]
+      p.sub = sub
+      p.name = name
+      p.value = value
+      p
+    }
+
+    def unapply(l: SettingsRow): Option[(SystemName, String, String)] =
+      Some((l.sub, l.name, l.value))
+
+    val Zero: SettingsRow = apply(SystemName.ocs, "", "")
+  }
+
+  private def columns(p: Props): List[Table.ColumnArg] =
+    List(
+      Column(Column.props(p.size.width.toInt / 2, "name", label = "Name")),
+      Column(Column.props(p.size.width.toInt / 2, "value", label = "Value"))
+    )
+
+  def rowClassName(p: Props)(i: Int): String = ((i, p.rowGetter(i)) match {
+    case (-1, _)                                                   =>
+      SeqexecStyles.headerRowStyle
+    case (_, SettingsRow(s, _, _)) if s === SystemName.instrument =>
+      SeqexecStyles.stepRow + SeqexecStyles.rowPositive
```

Comment:
I assume `+` operations here just merge the two styles together?

Reply:
That is correct